--- valgrind-3.5.0-orig/configure	2009-08-19 16:44:07.000000000 +0300
+++ valgrind-3.5.0/configure	2009-08-23 14:17:58.000000000 +0300
@@ -1819,6 +1821,7 @@
 #----------------------------------------------------------------------------
 # Checks for various programs.
 #----------------------------------------------------------------------------
+ac_save_CFLAGS=$ac_env_CFLAGS_value
 CFLAGS="-Wno-long-long"
 
 echo "$as_me:$LINENO: checking whether ln -s works" >&5
@@ -2428,7 +2431,7 @@
 echo "${ECHO_T}$ac_cv_c_compiler_gnu" >&6
 GCC=`test $ac_compiler_gnu = yes && echo yes`
 ac_test_CFLAGS=${CFLAGS+set}
-ac_save_CFLAGS=$CFLAGS
+ac_save_CFLAGS="$ac_save_CFLAGS $CFLAGS"
 CFLAGS="-g"
 echo "$as_me:$LINENO: checking whether $CC accepts -g" >&5
 echo $ECHO_N "checking whether $CC accepts -g... $ECHO_C" >&6
